.github/actions/write-hash/action.yml

```
@@ -0,0 +1,28 @@
+name: 'Write hash to file'
+description: 'Write a hash to a file and push it to the CI branch'
+inputs:
+  hash:
+    description: 'The hash to be written to file'
+    required: true
+    default: ''
+  file:
+    description: 'The filename (relative path from top level directory)'
+    required: true
+    default: ''
+  ci_branch:
+    description: 'The name of the CI branch'
+    required: false
+    default: 'ci'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Write hash to file
+      run: |
+        git config user.name github-actions
+        git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+        git fetch && git checkout ${{ inputs.ci_branch }} && git pull && git rebase origin/main
```

You can avoid a global fetch and subsequent pull by fetch-pulling the `ci` branch specifically.
```suggestion
        git fetch origin ${{ inputs.ci_branch }}:${{ inputs.ci_branch }} && git checkout ${{ inputs.ci_branch }} && git rebase origin/main
```